modin/backends/base/query_compiler.py

```diff
@@ -2066,7 +2066,7 @@ def inserter(df, loc, column, value):
             df.insert(loc, column, value)
             return df
 
-        return DataFrameDefault.register(inserter)(
+        return DataFrameDefault.register(inserter, inplace=True)(
```

Comment:
Since insert is an in-place operation we should return an object to which function was applied but not the result of the function. Even though we define inserter above we should always keep things explicit.